diff --git a/net/socket.c b/net/socket.c
index 976426d03f09..f2217169c668 100644
--- a/net/socket.c
+++ b/net/socket.c
@@ -1797,6 +1797,7 @@ int __sys_accept4_file(struct file *file, unsigned file_flags,
 int __sys_accept4(int fd, struct sockaddr __user *upeer_sockaddr,
 		  int __user *upeer_addrlen, int flags)
 {
 	int ret = -EBADF;
+	printk("Inside Accept Kernel");
 	struct fd f;
 
@@ -1859,6 +1860,8 @@ int __sys_connect_file(struct file *file, struct sockaddr_storage *address,
 
 int __sys_connect(int fd, struct sockaddr __user *uservaddr, int addrlen)
 {
 	int ret = -EBADF;
+	printk("Inside Connect Kernel");
 	struct fd f;
 
